### DISCUSSION TITLE
Get `route` from App Router

### DISCUSSION THREAD

**Discussion by giannif** [vercel/next.js#53398] [App Router]
I'm looking for a replacement for `route` in the App Router. 

Using `next/router` in the pages directory I would use this: 

```ts
import { useRouter } from 'next/router'

const { route } = useRouter() 
```

And route would be a string with something like `search/[term]`. 

The docs don't mention the `route` prop in the [migration guide](https://nextjs.org/docs/app/api-reference/functions/use-router#userouter). 

Is this on the roadmap or will be support be dropped? Or is there an alternative?

**Comment by v3rron** [comment]
Same here. I posted about the same [issue#53461](https://github.com/vercel/next.js/discussions/53461) with real world example provided.

**Comment by Uy575** [comment]
you are importing useRouter from next/router in latest update it is replace with navigation import useRouter from next/navigation instead of next/router may be it solves your problem

**Reply by giannif** [reply]
No not even close
